components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/extension/eidas/EidasExtensionProcessor.java

```diff
@@ -153,8 +154,18 @@ public void processSAMLExtensions(StatusResponseType response, Assertion asserti
                 log.debug("Process and validate a response against the SAML request with extensions" +
                         " for EIDAS message format");
             }
+            assertion.setSignature(null);
```

Comment:
This has been done due to following restrictions.
1) All the extension processors are invoked after building the assertion and before encrypting as in [1], since there are few modifications to the assertion in eIDAS processor.
2) In default saml assertion builder in [2], returns the signed assertion. Since there will be modifications to the assertion in eidas extension processor, assertion will be signed again after those modifications.
3) Extension processor invocation can't move inside the  default saml assertion builder (i.e. before signing), since the response object should be needed for eidas extension processor to build and send a error response 

[1] https://github.com/wso2-extensions/identity-inbound-auth-saml/blob/342cb8d5fc1ea2f1fff5f309379902db6e2f23eb/components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/builders/DefaultResponseBuilder.java#L94
[2] https://github.com/wso2-extensions/identity-inbound-auth-saml/blob/342cb8d5fc1ea2f1fff5f309379902db6e2f23eb/components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/builders/assertion/DefaultSAMLAssertionBuilder.java#L103

Reply:
Shall we create a issue indicating the limitation here.

Ideally, we should perform any modification to the assertion before it gets signed.
Signing should be the very last thing to do

Reply:
Created an issue for this in https://github.com/wso2/product-is/issues/3455